### PR TITLE
Update allocations information

### PR DESCRIPTION
**Pull request by nanobash**
## Summary

[Preview](https://preview.redoc.ly/rebilly-core/update-allocations-delete-information/)

- [x] Add allocations information
    - To delete transaction allocations, send an empty transaction allocation array in the request.
    - To modify transaction allocations, send a modified transaction allocation array in the request.
    - If you do not want to modify or delete transaction allocations, do not send a transaction allocation array in the request.

## Checklist

- [x] Writing style
- [x] API design standards

### CODE REVIEW

File: openapi/components/schemas/CreditMemos/CommonCreditMemo.yaml [hunk 1]
```diff
@@ -17,6 +17,10 @@ properties:
         type: array
         description: |-
           List of transactions (typically refunds, credits, and chargebacks) that are allocated to reduce the unused amount of a credit memo.
+
+            - To delete transaction allocations, send an empty transaction allocation array.
+            - To modify transaction allocations, do send a modified transaction allocation array.
+            - Not to modify transaction allocations, do not send a transaction allocation array.
```
Comment (noon-dawg) [inline openapi/components/schemas/CreditMemos/CommonCreditMemo.yaml:23]: nitpick: suggestion for readability
```suggestion
            - To delete transaction allocations, send an empty transaction allocation array in the request.
            - To modify transaction allocations, send a modified transaction allocation array in the request.
            - If you do not want to modify transaction allocations, do not send a transaction allocation array in the request.
```

File: openapi/components/schemas/CreditMemos/CommonCreditMemo.yaml [hunk 2]
```diff
@@ -42,7 +46,13 @@ properties:
...
       invoices:
         type: array
-        description: List of invoices that the credit memo is allocated to.
+        description: |-
+          List of invoices that the credit memo is allocated to.
+
+          - To delete invoice allocations, send an empty invoice allocation array.
+            Paid invoices cannot be modified or deleted.
+          - To modify invoice allocations, do send a modified invoice allocation array.
+          - Not to modify invoice allocations, do not send an invoice allocation array.
```
Comment (noon-dawg) [inline openapi/components/schemas/CreditMemos/CommonCreditMemo.yaml:55]:
```suggestion
          - To delete invoice allocations, send an empty invoice allocation array in the request.
            Paid invoices cannot be modified or deleted.
          - To modify invoice allocations, send a modified invoice allocation array in the request.
          - If you do not want to modify invoice allocations, do not send an invoice allocation array in the request.
```

File: openapi/components/schemas/CreditMemos/CommonCreditMemo.yaml (hunk 1 above)
Comment (nanobash) [reply openapi/components/schemas/CreditMemos/CommonCreditMemo.yaml]: Updated, thanks

File: openapi/components/schemas/CreditMemos/CommonCreditMemo.yaml (hunk 2 above)
Comment (nanobash) [reply openapi/components/schemas/CreditMemos/CommonCreditMemo.yaml]: Updated, thanks

File: openapi/components/schemas/CreditMemos/CommonCreditMemo.yaml [hunk 3]
```diff
@@ -17,6 +17,10 @@ properties:
         type: array
         description: |-
           List of transactions (typically refunds, credits, and chargebacks) that are allocated to reduce the unused amount of a credit memo.
+
+            - To delete transaction allocations, send an empty transaction allocation array in the request.
+            - To modify transaction allocations, send a modified transaction allocation array in the request.
+            - If you do not want to modify transaction allocations, do not send a transaction allocation array in the request.
```
Comment (nanobash) [inline openapi/components/schemas/CreditMemos/CommonCreditMemo.yaml:23]:
```suggestion
            - If you do not want to modify or delete transaction allocations, do not send a transaction allocation array in the request.
```

File: openapi/components/schemas/CreditMemos/CommonCreditMemo.yaml [hunk 4]
```diff
@@ -42,7 +46,13 @@ properties:
...
       invoices:
         type: array
-        description: List of invoices that the credit memo is allocated to.
+        description: |-
+          List of invoices that the credit memo is allocated to.
+
+          - To delete invoice allocations, send an empty invoice allocation array in the request.
+            Paid invoices cannot be modified or deleted.
+          - To modify invoice allocations, send a modified invoice allocation array in the request.
+          - If you do not want to modify invoice allocations, do not send an invoice allocation array in the request.
```
Comment (nanobash) [inline openapi/components/schemas/CreditMemos/CommonCreditMemo.yaml:55]:
```suggestion
          - If you do not want to modify or delete invoice allocations, do not send an invoice allocation array in the request.
```

File: openapi/paths/credit-memos.yaml [hunk 5]
```diff
@@ -33,7 +33,15 @@ get:
...
             items:
-              $ref: ../components/schemas/CreditMemos/CreditMemo.yaml
+              allOf:
+                - properties:
+                    allocations:
+                      properties:
+                        transactions:
+                          description: List of transactions (typically refunds, credits, and chargebacks) that are allocated to reduce the unused amount of a credit memo.
+                        invoices:
+                          description: List of invoices that the credit memo is allocated to.
```
Comment (ashkarpetin) [inline openapi/paths/credit-memos.yaml:43]: Why it can't be a part of credit memo schema?

Comment (nanobash) [reply openapi/paths/credit-memos.yaml:43]: Because for GET endpoints, it does not make sense to include extended information

Comment (adamaltman) [reply openapi/paths/credit-memos.yaml:43]: This doesn't make sense. First, it's missing type declarations.

Second, "for GET endpoints, it does not make sense to include extended information". Why does it not make sense? According to who?

**Review by adamaltman — CHANGES_REQUESTED** [review]
Incorrect API definitions and bad API design.

**Comment by nanobash** [comment]
> invoices

It does not make sense because GET endpoints does not accept request body, thus GET endpoints can't edit/delete

> Incorrect API definitions and bad API design.

I meant that GET endpoints does not accept request body, thus GET can't edit/delete allocations. I just wanted to be explicit in descriptions what endpoint can do what. But okay, we can go with the same descriptions for all the endpoints, if it's fine.

File: openapi/paths/credit-memos.yaml (hunk 5 above)
Comment (nanobash) [reply openapi/paths/credit-memos.yaml:43]: I meant that GET endpoints does not accept request body, thus GET can't edit/delete allocations. I just wanted to be explicit in descriptions what endpoint can do what. But okay, we can go with the same descriptions for all the endpoints, if it's fine.

Comment (nanobash) [reply openapi/paths/credit-memos.yaml]: Updated, thanks

**Review by adamaltman — APPROVED** [review]

**Comment by nanobash** [comment]
@Rebilly/php please review

**Review by ashkarpetin — APPROVED** [review]

**Pull request merged**